net-misc/kea/files/kea-1.8.2-gtest.patch

```diff
@@ -0,0 +1,141 @@
+-            [enable_gtest="yes" ; GTEST_SOURCE="$withval"])
+-
+-AC_ARG_WITH([gtest],
+-            [AS_HELP_STRING([--with-gtest=PATH],
+-                            [specify a path to gtest header files (PATH/include) and library (PATH/lib)])],
+-        [gtest_path="$withval"; enable_gtest="yes"], [gtest_path="no"])
++AC_ARG_ENABLE([gtest], [AS_HELP_STRING([--enable-gtest],
++        [enable testing via Googletest [default=no]])],
++        enable_gtest="yes"; enable_gtest="no")
+
+ AC_ARG_WITH([lcov],
+             [AS_HELP_STRING([--with-lcov=PROGRAM],
+@@ -55,110 +47,22 @@
+ #
+ # Check availability of gtest, which will be used for unit tests.
+ #
+-GTEST_LDFLAGS=
+-GTEST_LDADD=
+ DISTCHECK_GTEST_CONFIGURE_FLAG=
+-GTEST_VERSION="unknown"
+
+ if test "x$enable_gtest" = "xyes" ; then
+-
+-    DISTCHECK_GTEST_CONFIGURE_FLAG="--with-gtest=$gtest_path"
+-
+-    if test -n "$with_gtest_source" ; then
+-
+-        if test "x$GTEST_SOURCE" = "xyes" ; then
+-
+-            AC_MSG_CHECKING([for gtest source])
+-            # If not specified, try some common paths.
+-            GTEST_SOURCE=
+-            for d in /usr/src/gtest /usr/local /usr/pkg /opt /opt/local ; do
+-                if test -f $d/src/gtest-all.cc -a $d/src/gtest_main.cc; then
+-                    GTEST_SOURCE=$d
+-                    AC_MSG_RESULT([$GTEST_SOURCE])
+-                    break
+-                fi
+-            done
+-            if test -z $GTEST_SOURCE ; then
+-                AC_MSG_ERROR([no gtest source but it was selected])
+-            fi
+-        else
+-            if test ! -d $GTEST_SOURCE/src -a -d $GTEST_SOURCE/googletest; then
+-                GTEST_SOURCE=$GTEST_SOURCE/googletest
+-            fi
+-            if test -f $GTEST_SOURCE/src/gtest-all.cc -a $GTEST_SOURCE/src/gtest_main.cc; then
+-                have_gtest_source=yes
+-            else
+-                AC_MSG_ERROR([no gtest source at $GTEST_SOURCE])
+-            fi
+-        fi
+-        have_gtest_source=yes
+-
+-        GTEST_LDADD="\$(top_builddir)/ext/gtest/libgtest.a"
+-        DISTCHECK_GTEST_CONFIGURE_FLAG="--with-gtest-source=$GTEST_SOURCE"
+-        GTEST_INCLUDES="-I$GTEST_SOURCE -I$GTEST_SOURCE/include"
+-        GTEST_VERSION="`basename $GTEST_SOURCE`"
+-
+-# Versions starting from 1.8.0 are put in the googletest directory. If the basename
+-# returns googletest string, we need to cut it off and try baseline again.
+-        if test "$GTEST_VERSION" == "googletest"; then
+-            GTEST_VERSION=${GTEST_SOURCE%"/googletest"}
+-            GTEST_VERSION=`basename $GTEST_VERSION`
+-        fi
+-        GTEST_VERSION="${GTEST_VERSION#googletest-release-}"
+-        GTEST_VERSION="${GTEST_VERSION#gtest-}"
+-    fi
++    DISTCHECK_GTEST_CONFIGURE_FLAG="--with-gtest"
++    PKG_CHECK_MODULES([GTEST], [gtest], [], [AC_MSG_ERROR([gtest requested but not found])])
++    GTEST_INCLUDES=`pkg-config --keep-system-cflags --cflags-only-I gtest`
```

Comment:
You shouldn't need to then call pkg-config directly (PKG_CHECK_MODULES should populate everything you need), but if you absolutely must, respect ${PKG_CONFIG} instead (not calling just pkg-config bare).

Reply:
@thesamesam thank you for the hint with ${PKG_CONFIG} i replaced  the plain `pkg-config`calls. Sadly PKG_CHECK_MODULES does not provide everything which is required by kea. I also wanted to keep the changes minimal on their codebase.

Reply:
np, I understand -- thanks! I defer to @Polynomial-C now.